lib/pure/decimal.nim

```diff
@@ -0,0 +1,1040 @@
+##[
+  The ``decimal`` module supports the storage of numbers in decimal format.
+
+  The primary benefit of this is it avoids conversion errors when converting to/from
+  decimal (base 10) and binary (base 2). The is critical for applications where
+  the numbers are start out as decimal, must be output as decimal, and even minor
+  error are problematic. Financial and scientific lab programs, for example, often
+  meet this requirement.
+
+  As a secondary benefit, this library also honors the "significance" of a number
+  and properly handles significance during mathematic operations.
+
+  Examples
+  ========
+  .. code-block:: nim
+    import decimal
```

Comment:
Prefer `runnableExamples` over `.. code-block:: nim` (they also automatically add the module import).

Reply:
fixed in next commit.